### DISCUSSION TITLE
Technical issues?

### DISCUSSION THREAD

**Discussion by jllyh** [jlai/fitness-dashboard#64] [General]
It seems that only my sleep and heart rate values are loading today.  All other stats show as **"Error loading: NetworkError when attempting to fetch resource."** Interestingly, the heart rate values appear to be current and updating, so there is still some connection.

**Comment by jlai** [comment]
Seems to be working for me right now. Is it still happening for you? One other user reported this, maybe some temporary Fitbit API outage?

**Reply by jllyh** [reply]
Still happening. This is what I'm seeing in my usual Firefox browser: 
![image](https://github.com/user-attachments/assets/e81fbf8b-d6b7-4017-8ade-ba5fafe72be0)
I just tried Chrome and Edge as well, and I get similar results.

As I noted, what strikes me as curious is that the heart rate is populating and updating as expected. 
![image](https://github.com/user-attachments/assets/efc90a4d-eaee-4c2a-abe6-1109eb05cc86)

**Reply by TLeafTO** [reply]
same here!

**Reply by JimmyG1814** [reply]
Having the problem as well

**Comment by jlai** [comment]
In Chrome, could you open up the developer console (Control-Shift-J or "..." menu -> More Tools -> Developer Tools) and take a screenshot of the error logs?

**Reply by jllyh** [reply]
![image](https://github.com/user-attachments/assets/11dd4b7f-1a2e-4011-aad0-f1ee9f6b03fa)

**Reply by jllyh** [reply]
For what it's worth, here is the take from Firefox.
![image](https://github.com/user-attachments/assets/f4966ece-e909-4836-bd95-be7a98d5c622)

**Comment by jlai** [comment]
Thanks. Unfortunately the error message doesn't tell me much. Are you able to load other pages besides the dashboard, like the activity/sleep history pages?

**Reply by jllyh** [reply]
One more bit, from Network under the GET statement that's in error:
![image](https://github.com/user-attachments/assets/2ddb7084-1e4d-4a6d-abab-93eccf495b66)
As for your question, in History only the Sleep and Weight logs appear to be loading. I'm not sure what to expect in Activities (that's empty) and Steps do not load for any date range (even when I use Custom to back today's date out of the range).

**Reply by jllyh** [reply]
For the drop-down in History, this seems to be the pattern. If it's not working, it's red. If it IS working, it's green. If it's unmarked, I either don't use it or don't know what to expect.
![image](https://github.com/user-attachments/assets/90a901af-db17-42f2-a0b0-2e7dab29a044)

**Comment by jlai** [comment]
The cloudflare one is just for stats tracking (counting how many users are using the website). OK if that one is blocked.

What I'd be looking for is on the Network tab of the inspector, errors for api.fitbit.com. Choose a GET request, not OPTIONS. Look for the Response Headers (don't screenshot the Request Headers because it has login/authorization tokens) and the Response tab (enable Raw checkbox).

![network](https://github.com/user-attachments/assets/c15a66ab-72ee-4707-b715-f9111c9e758e)

(note, there may be a couple of them with the date in the URL, hover over them and pick the one that has "steps" not "sleep" or other)

![network-response](https://github.com/user-attachments/assets/e3c9cb3d-f33c-4eef-98b9-773e35661e99)
![network-response-tab](https://github.com/user-attachments/assets/77a83a76-2ed0-43cd-a5d5-5af716ee7162)

**Reply by jllyh** [reply]
Edit - I'm going to try that again with a longer range (just one day in September).

I should have left it as it was. This is what I had with the default of September; the only api.fitbit.com GET statement showing an error (and it does involve steps):
![image](https://github.com/user-attachments/assets/ddbd2405-feaa-469a-9c80-d24c7e83cacb)

DAY and WEEK return data for just one day (it's the first of the month and a Sunday, so those fit). But MONTH and YEAR don't even return that one day of data. And in all cases, the GET error is returned and the dashboard display does not update.
![image](https://github.com/user-attachments/assets/0d4d8a02-38a3-4789-b950-8d1072c72ada)

**Reply by jlai** [reply]
Could you screenshot the Response section of the Headers tab (as mentioned, crop out the Request section)? It seems like the Fitbit server is not sending one of the required headers for some reason -- but not sure why it's happening on your account but not mine.

**Reply by jllyh** [reply]
Sorry, I missed that.
![image](https://github.com/user-attachments/assets/367b9ef5-348b-4ebe-92e6-6fd6879275bd)

**Reply by jllyh** [reply]
Ah, I had RAW enabled for that but you hadn't requested it. Second attempt:
![image](https://github.com/user-attachments/assets/88175ca9-9d59-497e-88f0-2f9e88c9a1a3)

**Reply by jlai** [reply]
Thanks. So this seems to be some kind of issue with the Fitbit API so I'll need to contact their developer support. Unfortunately since it's a holiday weekend in the US, I don't expect a response for a few days.

**Reply by jllyh** [reply]
Thanks for the good work. Here's hoping for a swift resolution.

**Reply by jlai** [reply]
Thanks for helping investigate with the screenshots!

**Comment by Carmapet** [comment]
Same issue for me when I opened it up.

**Comment by TLeafTO** [comment]
Same here. It started a few days ago. Chrome/Windows 10
![image](https://github.com/user-attachments/assets/879d3dea-6366-4a43-9e8e-3fca045f4b53)

**Comment by jlai** [comment]
Seems to be affecting all users now. Most likely, they made some internal code changes which they didn't realize would break 3rd party apps, and then rolled it out incrementally.

**Reply by TLeafTO** [reply]
I did notice it happening about three days ago but it then resolved same day.

**Comment by mscatz-9** [comment]
Looks like it's impacting all 3rd party apps.  [https://community.fitbit.com/t5/Third-Party-Integrations/bd-p/other-apps](url)  as well as incorrect information on the Fitbit app.

**Reply by jllyh** [reply]
I'm not able to follow the link (404 error) but I'll take your word for it.
Ah ... the text in your link is different than the link itself. 
[https://community.fitbit.com/t5/Third-Party-Integrations/bd-p/other-apps](https://community.fitbit.com/t5/Third-Party-Integrations/bd-p/other-apps)

**Comment by jllyh** [comment]
I'm seeing steps data!  There are still several unpopulated areas of the dashboard, but it's the first improvement in days. Here's hoping that it all gets resolved soon.

**Reply by jllyh** [reply]
I'm still seeing several 500 errors. Here's hoping that those resolve themselves soon.
-edit - As of 6:47 EDT, resolved!

**Comment by TLeafTO** [comment]
Yay!!!! Fitness Dashboard is back! For me at least and I hope for you too! 
![image](https://github.com/user-attachments/assets/447ed227-2c92-426c-9f35-8dcc264b2674)

**Reply by mscatz-9** [reply]
Back for me too.  :-)

**Comment by TLeafTO** [comment]
Oh no, it's down again :(
![image](https://github.com/user-attachments/assets/c73b40e2-e408-4467-a096-b71df30497e7)

**Reply by jlai** [reply]
401 is a different kind of error, usually a problem with the authentication token. Logging out and in again might help if you see that.

**Comment by TLeafTO** [comment]
..and now it's back again :)
Seems a little unstable?
![image](https://github.com/user-attachments/assets/d3b5cb45-d802-4a28-a565-a5eb1c6e54ca)
